### ISSUE TITLE
Error when trying to run on test dataset

### ISSUE THREAD

**Issue by papelypluma** [open2c/distiller-nf#161]
Hi distiller-nf developers. I'm a novice user of singularity and docker who's attempting to run distiller-nf on the sample/test dataset. I'm following the test example instructions, but I'm getting the following error: 

> Error executing process > 'download_truncate_chunk_fastqs (library:MATalpha_R1 run:lane2)'
> 
> Caused by:
>   Cannot run program "bsub" (in directory "/home/user001/tools/distiller/mirniylab_distiller-nf/work/73/fc2d32d01b5e7e8fc8d53a40a3e38c"): error=2, No such file or directory
> 
> Command executed:
> 
>   bsub
> 
> Command exit status:
>   -

I modified cluster.config setting singularity enabled to true and autoMounts to true as well. Since our server doesn't have docker installed but singularity instead, I invoked the command as follows: nextflow distiller.nf -params-file ./test/test_project.yml -profile cluster

Any help on this matter will be highly appreciated. Thanks!

**Comment by Phlya** [comment]
Did you set the correct executor in the cluster config? How do you submit jobs normally on the cluster, is it using `bsub`?

**Comment by papelypluma** [comment]
Hi @Phlya. Thanks for the response. That's what I didn't set. We're using SLURM to run schedule jobs on the server. Is that the only thing that has to be changed or set before running distiller? Since I'm testing it on the test dataset, I'm wondering if there's a way to bypass the need to submit the job through a scheduler? 

Thanks!

**Comment by Phlya** [comment]
Then don't use the cluster config, just use the local profile, and do the necessary modifications there.

**Comment by papelypluma** [comment]
Thanks for the help. I tried the local profile, but then an error came out:

> Error executing process > 'map_parse_sort_chunks (library:MATalpha_R1 run:lane2 chunk:0)'
> Caused by:
>   Process `map_parse_sort_chunks (library:MATalpha_R1 run:lane2 chunk:0)` terminated with an error exit status (127)

I set docker enabled to false.

**Comment by Phlya** [comment]
Um without docker, do you actually have all the tools installed? Why did you disable it?

**Comment by papelypluma** [comment]
We don't have docker currently installed, but we have singularity instead.

**Comment by Phlya** [comment]
Right, so did you use singularity?

**Comment by papelypluma** [comment]
I tried to copy the singularity lines from cluster.config, but it also resulted to an error. I'm trying to figure out how to specify singularirty in the local.config.

**Comment by Phlya** [comment]
OK, I've never used either, so don't think I can help with that...

**Comment by Phlya** [comment]
Instead of docker you can just create a conda environment with all the right dependencies, by the way.

**Comment by papelypluma** [comment]
Hi @Phlya. Thanks for the suggestions and help. It's harder than I thought, but I'll keep trying. Will also try your suggestions.

**Comment by papelypluma** [comment]
Just sharing an update. I finally managed to make it work using singularity with local config. I'm not sure if the docker image has fastp in it to begin with, but when I pulled and converted mirnylab/distiller_env:0.3.3 to simg file fastp was already missing (error points to something like `fastp` command cannot be found). I resolved it by binding a `fastp` binary to /miniconda3/bin (in the local config file for singularity) as follows: `runOptions = "--bind `pwd`:/mnt:rw -B /absolute/path/to/fastp:/miniconda3/bin/fastp"`.
